internal/hcs/process.go

```diff
@@ -197,15 +198,65 @@ func (process *Process) Kill() (_ bool, err error) {
 func (process *Process) waitBackground() {
 	operation := "hcsshim::Process::waitBackground"
 	process.logOperationBegin(operation)
-	err := waitForNotification(process.callbackNumber, hcsNotificationProcessExited, nil)
+
+	var (
+		err      error
+		exitCode = -1
+	)
+
+	err = waitForNotification(process.callbackNumber, hcsNotificationProcessExited, nil)
 	if err != nil {
-		err = makeProcessError(process, "Wait", err, nil)
+		err = makeProcessError(process, operation, err, nil)
+		logrus.WithFields(logrus.Fields{
+			logfields.ContainerID: process.SystemID(),
+			logfields.ProcessID:   process.processID,
+			logrus.ErrorKey:       err,
+		}).Errorf("%s - failed wait", operation)
+	} else {
+		process.handleLock.RLock()
+		defer process.handleLock.RUnlock()
+
+		// Make sure we didnt race with Close() here
+		if process.handle != 0 {
+			var (
+				resultp     *uint16
+				propertiesp *uint16
+			)
+			err = hcsGetProcessPropertiesContext(gcontext.TODO(), process.handle, &propertiesp, &resultp)
```

Comment:
Should this be gcontext.Background() now or what?

Reply:
In the OC PR I use the `waitBackground` `span` which is `context.Background` yes. I can do that now and update if you want. It seemed appropriate to use `TODO()` since it should be updated.

Reply:
OK just wanted to understand what the TODO was here. Either way's fine as long as it doesn't linger as TODO.

Reply:
Yep agreed.